packages/block-library/src/navigation/edit/index.js

```diff
@@ -283,7 +297,24 @@ function Navigation( {
 		hasResolvedCanUserCreateNavigationMenu,
 	} = useNavigationMenu( ref );
 
+	// Attempt to retrieve and prioritize any existing navigation menu unless
+	// a specific ref is allocated or the user is explicitly creating a new menu. The aim is
+	// for the block to "just work" from a user perspective using existing data.
+	useEffect( () => {
+		if (
+			isCreatingEmpty.current ||
```

Comment:
Does it have to be specifically `isCreatingEmpty`? As a user, I would expect this to short-circuit if this block's `isCreatingNavigationMenu` is true in general.

Reply:
🤔

Reply:
I'm sure I tried this approach. Let me see...

Reply:
This is very odd. I'm sure I tried that and there was a problem. I mean you're right it seems obvious.

I've just tried it again and now it seems to be working fine 🥳